File: app/src/main/java/net/numa08/gochisou/presentation/internal/di/modules/RealmConfigurationModule.java
package net.numa08.gochisou.presentation.internal.di.modules;

import android.content.Context;

import net.numa08.gochisou.presentation.internal.di.PerActivity;

import dagger.Module;
import dagger.Provides;
import io.realm.RealmConfiguration;

@Module
public class RealmConfigurationModule {

    @Provides
    @PerActivity
    public RealmConfiguration providesRealmConfiguration(Context context) {
        return new RealmConfiguration.Builder(context.getCacheDir())
                .name("esa")
                .deleteRealmIfMigrationNeeded()
                .build();
    }

}

File: app/src/main/kotlin/net/numa08/gochisou/presentation/internal/di/modules/RealmConfigurationModule.kt
package net.numa08.gochisou.presentation.internal.di.modules

import android.content.Context
import dagger.Module
import dagger.Provides
import io.realm.RealmConfiguration
import net.numa08.gochisou.presentation.internal.di.PerActivity

@Module
class RealmConfigurationModule {

    @Provides
    @PerActivity
    fun providesRealmConfiguration(context: Context): RealmConfiguration {
        return RealmConfiguration.Builder(context.cacheDir).name("esa").deleteRealmIfMigrationNeeded().build()
    }

}
